### PR TITLE
Add: slackbot will re-create a conversation if needed, url is now visibile to the model.

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

This PR enhances the Slackbot functionality:
- Adds ability to re-create a conversation if needed
- Makes the URL visible to the model

## Risk

Low risk. Changes are confined to Slackbot conversation handling.

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: connectors/src/connectors/slack/bot.ts [hunk 1]
```diff
@@ -593,34 +593,62 @@ async function answerMessage(
...
     return buildSlackMessageError(buildContentFragmentRes);
   }
+
   let conversation: ConversationType | undefined = undefined;
   let userMessage: UserMessageType | undefined = undefined;
+
   if (lastSlackChatBotMessage?.conversationId) {
-    if (buildContentFragmentRes.value) {
-      const contentFragmentRes = await dustAPI.postContentFragment({
+    const ConversationNotFoundError = class extends Error {
```
Comment (flvndvd) [inline connectors/src/connectors/slack/bot.ts:601]: No classes/types within functions body.

Also why a custom error classes here?

Comment (flvndvd) [reply connectors/src/connectors/slack/bot.ts:601]: Also you can do something simpler for customer error classes.

```
const ConversationNotFoundError = class extends Error {
    readonly name = "ConversationNotFoundError";

    constructor(message: string) {
        super(message);
      }
}
```

File: connectors/src/connectors/slack/bot.ts [hunk 2]
```diff
@@ -593,34 +593,62 @@ async function answerMessage(
...
+        conversationId: lastSlackChatBotMessage.conversationId,
+      });
+      if (conversationRes.isErr()) {
+        if (conversationRes.error.type === "conversation_not_found") {
+          throw new ConversationNotFoundError("Conversation not found");
+        }
+        return buildSlackMessageError(conversationRes);
+      }
+      conversation = conversationRes.value;
+    } catch (ConversationNotFoundError) {
```
Comment (flvndvd) [inline connectors/src/connectors/slack/bot.ts:645]: In TS you can't capture explicit error types. Here we basically swallow all the errors which is definitely not what we want 🙃.

File: connectors/src/connectors/slack/bot.ts [hunk 3]
```diff
@@ -593,34 +593,62 @@ async function answerMessage(
...
+
+    try {
+      if (buildContentFragmentRes.value) {
+        const contentFragmentRes = await dustAPI.postContentFragment({
+          conversationId: lastSlackChatBotMessage.conversationId,
+          contentFragment: buildContentFragmentRes.value,
+        });
+        if (contentFragmentRes.isErr()) {
+          if (contentFragmentRes.error.type === "conversation_not_found") {
+            throw new ConversationNotFoundError("Conversation not found");
```
Comment (flvndvd) [inline connectors/src/connectors/slack/bot.ts:617]: Not a big fan of this pattern.

File: connectors/src/connectors/slack/bot.ts [hunk 4]
```diff
@@ -593,34 +593,48 @@ async function answerMessage(
...
-      conversationId: lastSlackChatBotMessage.conversationId,
-    });
-    if (conversationRes.isErr()) {
-      return buildSlackMessageError(conversationRes);
-    }
-    conversation = conversationRes.value;
-  } else {
+  }
+
+  if (!conversation || !userMessage) {
```
Comment (flvndvd) [inline connectors/src/connectors/slack/bot.ts:637]: The check on conversation should be enough, right?

**Review by flvndvd — APPROVED** [review]

File: connectors/src/connectors/slack/bot.ts (hunk 4 above)
Comment (Fraggle) [reply connectors/src/connectors/slack/bot.ts:637]: typescript complains :-)

**Pull request merged**